aws-observability/aws-application-signals-test-framework
# Add Test Cases for Metric Rollup Addition of [RemoteService, RemoteTarget, ...]

*Issue #, if available:*
https://github.com/aws/amazon-cloudwatch-agent/pull/1010
Recently the following metric rollups were added to 2 AppSignal Configuration files. Tests need to be added for them
  - `[RemoteService, RemoteTarget]`
  - `[HostedIn.<Attributes>, Service, RemoteService, RemoteTarget]`

*Description of changes:*

Add tests for new metric rollups
- Update Java EKS/EC2 e2e tests workflows
  - Provide  `testingId=${{ env.TESTING_ID }}` to the sample app via request-body to pass new Metric Validations
- Update Validator and validations for both Java and Python
  - Metric validation test cases now checks for a new S3 `RemoteTarget` value that includes testingId
    - (`::s3:::e2e-test-bucket-name-{{testingId}}`)
  - Add new test cases for the new metric rollups
  - Update metric validator to be able to filter against multiple dimensions
  - Query metrics that have both `RemoveService="AWS.SDK.S3"` and `RemoteTarget=<s3_bucket_name_input>`

*Testing done:*
15 minute canaries running successfully for 12+ hours:
EKS in IAD: https://github.com/jj22ee/aws-application-signals-test-framework/actions/workflows/test-eks-java.yml
EC2 in IAD: https://github.com/jj22ee/aws-application-signals-test-framework/actions/workflows/test-ec2-java.yml

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
